science/hdf4/Portfile

```diff
@@ -69,6 +70,8 @@ if {[fortran_variant_isset]} {
     configure.args-delete   --disable-fortran --enable-shared
 }
 
+compilers.allow_arguments_mismatch  yes
```

Comment:
Out of interest what does this option do ?

Reply:
https://github.com/macports/macports-ports/blob/73613c0a068534b30ea660189ddd8f9eae46df86/_resources/port1.0/group/compilers-1.0.tcl#L49

Reply:
The following Fortran code should be an error
```
program main
  print *, my_function(1)
  print *, my_function(1.0)
end program main
```

Newer version of GCC (10+) do not allow this.
With `-fallow-argument-mismatch` or on older version of GCC, it compiles (perhaps with a warning).

Because Fortran passes arguments by reference,  programers sometimes blur the distinction between, e.g., an integer and an array of integers.
There also seem to be clever ways of manipulating pointers/references/arrays when interfacing with C/C++ code.
I do not present to fully understand the details.

When arguments mismatches started becoming an error, `-fallow-argument-mismatch` (https://github.com/macports/macports-ports/commit/20b350bb5eb864e9323f2f79324a2f7f977942d1)
allowed a bridge for ports when a proper solution was not evident.